### PR TITLE
Rename auth.enabled to multitenancy.enabled

### LINKED ISSUE #639: Non-multi-tenant mode.
👋🏽 Is there a mode without multi-tenancy as designed by tempo? I have simple demo app which literally uses OTLP gRPC on clients and Tempo receiver as:

```
 server:
      http_listen_port: 9090
    ingester:
      trace_idle_period: 10s
      max_block_duration: 5m0s
      max_block_bytes: 1000000
    distributor:
      receivers:
        otlp:
          protocols:
            grpc:
              endpoint: 0.0.0.0:9091
    compactor:
      compaction:
        compaction_window: 1h0m0s
        max_block_bytes: 1000000
        block_retention: 1h0m0s
        compacted_block_retention: 10m0s
    storage:
      trace:
        pool:
          max_workers: 100
          queue_depth: 10000
        wal:
          path: /data/wal
          completedfilepath: ""
        block:
          index_downsample_bytes: 1048576
          index_page_size_bytes: 256000
          bloom_filter_false_positive: 0.05
          encoding: zstd
        backend: local
        local:
          path: /data/blocks
```

However, whenever my client (opentelemetry-go) pushes traces I got:

```
2021/04/10 22:50:01 rpc error: code = Unknown desc = no org id
2021/04/10 22:50:06 exporter disconnected
2021/04/10 22:50:11 exporter disconnected
2021/04/10 22:50:16 rpc error: code = Unknown desc = no org id
2021/04/10 22:50:21 exporter disconnected
2021/04/10 22:50:26 exporter disconnected
```

And on tempo:

```
imeout" ring=ingester
level=info ts=2021-04-10T22:46:10.051244922Z caller=app.go:224 msg="Tempo started"
level=info ts=2021-04-10T22:47:10.050428573Z caller=compactor.go:99 msg="enabling compaction"
level=info ts=2021-04-10T22:47:10.050569602Z caller=tempodb.go:370 msg="compaction and retention enabled."
level=error ts=2021-04-10T22:47:10.059266955Z caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
level=error ts=2021-04-10T22:47:10.059517872Z caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
level=error ts=2021-04-10T22:47:10.059684374Z caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
level=error ts=2021-04-10T22:47:10.059846183Z caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
level=error ts=2021-04-10T22:47:10.059852709Z caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
level=error ts=2021-04-10T22:48:19.002653074Z caller=log.go:27 msg="failed to extract org id" err="no org id"
level=error ts=2021-04-10T22:48:24.003988084Z caller=log.go:27 msg="failed to extract org id" err="no org id"
level=error ts=2021-04-10T22:48:34.003977664Z caller=log.go:27 msg="failed to extract org id" err="no org id"
level=error ts=2021-04-10T22:48:44.005104593Z caller=log.go:27 msg="failed to extract org id" err="no org id"
```

I think this is the problematic spot https://github.com/grafana/tempo/blob/master/modules/distributor/distributor.go#L204

Any ideas if there is a blocker to enable non tenancy mode on Tempo, or I have to somehow inject it?

### PR DESCRIPTION
**Pull request by dgzlopes**
**What this PR does**:
This PR renames `auth.enabled` to `multitenancy.enabled` and changes its default value to false.

**Which issue(s) this PR fixes**:
Fixes #639

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This section explains the configuration options for Tempo as well as the details
 Tempo uses the Weaveworks/common server.  See [here](https://github.com/weaveworks/common/blob/master/server/server.go#L45) for all configuration options.
 
 ```
-auth_enabled: false            # do not require X-Scope-OrgID
+auth_enabled: true            # optional. Require X-Scope-OrgID. By default it's set to false (disabled).
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/_index.md:24]:
```suggestion
auth_enabled: true            # Optional. Require X-Scope-OrgID. By default, it's set to false (disabled).
```

**Review by achatterjee-grafana — APPROVED** [review]
Minor change request, otherwise LGTM

File: docs/tempo/website/configuration/_index.md (hunk 1 above)
Comment (dgzlopes) [reply docs/tempo/website/configuration/_index.md]: Changed! Thanks :)

File: cmd/tempo/app/app.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func (c *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet) {
 	c.Target = All
 	// global settings
 	f.StringVar(&c.Target, "target", All, "target module")
-	f.BoolVar(&c.AuthEnabled, "auth.enabled", true, "Set to false to disable auth.")
+	f.BoolVar(&c.AuthEnabled, "auth.enabled", false, "Set to true to enable auth.")
```
Comment (joe-elliott) [inline cmd/tempo/app/app.go:65]: @grafana/tempo should we take this as an opportunity to improve this setting name?

`multitenancy.enabled` ?

Comment (annanay25) [reply cmd/tempo/app/app.go:65]: +1

Comment (joe-elliott) [reply cmd/tempo/app/app.go:65]: @dgzlopes Do you mind taking this on?

I think ideally we would actually support both and just mark `auth.enabled` deprecated and remove it from docs. A lot of people are going to have this setting in their config files.

Comment (dgzlopes) [reply cmd/tempo/app/app.go:65]: Yup, I'll refactor the PR :)

Comment (dgzlopes) [reply cmd/tempo/app/app.go]: Done! I added the new setting, marked the old one as deprecated, and removed a bunch of references to it.

**Review by joe-elliott — COMMENTED** [review]
We also pass `t.cfg.AuthEnabled` to the distributor when we create it. For sanity let's make a method `t.cfg.AuthEnabled()` or whatever and use it everywhere we used the old config value.

**Comment by dgzlopes** [comment]
Completely forgot about that @joe-elliott! :smile: 

Added the method and changed the references.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 * [CHANGE] `ingestion_rate_limit` limit override has been removed in favour of `ingestion_rate_limit_bytes`.
   `ingestion_burst_size` limit override has been removed in favour of `ingestion_burst_size_bytes`.
   This is a **breaking change** to the overrides config section. [#630](https://github.com/grafana/tempo/pull/630)
+* [CHANGE] Rename auth.enabled to multitenancy.enabled and set it as false by default. This is not a breaking change, but auth.enabled has been set as deprecated.
```
Comment (joe-elliott) [inline CHANGELOG.md:10]: Technically this is a breaking change if you were relying on auth/multitenancy being enabled by default. Let's list it here.

Also needs the `[#646](https://github.com/grafana/tempo/pull/646)` link

**Comment by joe-elliott** [comment]
Nice, tiny changelog fix other than that looks good. I will probably hold on merging this until we after we cut 0.7.0 since we haven't deployed/tested it anywhere.

File: CHANGELOG.md (hunk 3 above)
Comment (dgzlopes) [reply CHANGELOG.md]: Added both things to the changelog! :)

**Comment by joe-elliott** [comment]
@dgzlopes After :imp: #666 :imp: is merged you will need to rebase/merge main to get tests to pass.

**Comment by dgzlopes** [comment]
@joe-elliott Hopefully... nothing **hellish** happens :)

**Review by joe-elliott — APPROVED** [review]

**Comment by joe-elliott** [comment]
Seems like we're getting the same errors.  That commit fixed our other PRs. Can you double check again when you get a chance?

**Comment by dgzlopes** [comment]
I don't know what happened! It's working now @joe-elliott

**Pull request merged**